### PR TITLE
Baris/check integration

### PR DESCRIPTION
**Pull request by bcdurak**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] I have read the **CONTRIBUTING.md** document.
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

List of changes:
- Added a check to observe if the extra requirements are installed for each integration
- Updated the setup.py accordingly

### CODE REVIEW

File: zenml/utils/requirement_utils.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                  PYTORCH_INTEGRATION: PYTORCH_REQUIREMENTS,
+                  ALL_INTEGRATION: ALL_REQUIREMENTS}
+
+
+##################
+# UTIL FUNCTIONS #
+##################
+def check_integration(integration):
+    # Get the installed packages
+    reqs = subprocess.check_output([sys.executable, '-m', 'pip', 'freeze'])
```
Comment (htahir1) [inline zenml/utils/requirement_utils.py:69]: Do we need a subprocess? Cant we try and use importlib?

File: zenml/utils/requirement_utils.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    assert integration in EXTRAS_REQUIRE, \
+        f'At this moment, there is no integration for {integration}. ' \
+        f'Possible integrations for ZenML ' \
+        f'include: {list(EXTRAS_REQUIRE.keys())}.'
+
+    specs = EXTRAS_REQUIRE[integration]
+
+    for s in specs:
+        # Decouple from the version
+        pattern = r"([a-zA-Z0-9\-]+)(\[.+\])*(==(.+))*"
```
Comment (htahir1) [inline zenml/utils/requirement_utils.py:82]: What about weird patterns like >=, ~=, <= or something

File: zenml/utils/requirement_utils.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+AWS_REQUIREMENTS = []
+
+AZURE_INTEGRATION = 'azure'
+AZURE_REQUIREMENTS = []
+
+###################
+# EXTRAS: TOOLING #
+###################
+PYTORCH_INTEGRATION = 'pytorch'
+PYTORCH_REQUIREMENTS = ['torch']
```
Comment (htahir1) [inline zenml/utils/requirement_utils.py:45]: lets pin pytorch to >=1.7.0

**Review by htahir1 — COMMENTED** [review]
I like the idea of having the requirements in the module itself. Looks good to me - just some small changes. Well done!

File: zenml/utils/requirement_utils.py (hunk 3 above)
Comment (bcdurak) [reply zenml/utils/requirement_utils.py:45]: I'm on it!

File: zenml/utils/requirement_utils.py (hunk 1 above)
Comment (bcdurak) [reply zenml/utils/requirement_utils.py:69]: `importlib.utils.find_spec` failed to spot the `apache-beam` installation on my test. That's why I opted for an alternative solution.

File: zenml/utils/requirement_utils.py (hunk 2 above)
Comment (bcdurak) [reply zenml/utils/requirement_utils.py]: You are absolutely right! I have loosened the final group, as we are not currently using it!

File: zenml/utils/requirement_utils.py (hunk 1 above)
Comment (htahir1) [reply zenml/utils/requirement_utils.py:69]: You just need to replace the '-' with '_' i think

**Pull request merged**